pop-api/integration-tests/src/local_fungibles.rs

```diff
@@ -367,6 +403,69 @@ fn transfer_works() {
 	});
 }
 
+#[test]
+fn transfer_from_works() {
+	new_test_ext().execute_with(|| {
+		let _ = env_logger::try_init();
+		let addr = instantiate("contracts/fungibles/target/ink/fungibles.wasm", INIT_VALUE, vec![]);
+		let amount: Balance = 100 * UNIT;
+		let owner = addr.clone();
+		let delegate = CHARLIE;
+		let asset = ASSET_ID;
+
+		// Asset does not exist
+		assert_eq!(
+			decoded::<Error>(transfer_from(
+				addr.clone(),
+				asset,
+				delegate.clone(),
+				ALICE,
+				BOB,
+				1,
+				vec![]
+			)),
+			Module { index: 52, error: 3 },
+		);
+
+		// Allow `delegate` to spend `amount` owned by contract address
+		let asset = create_asset_and_mint_to(ALICE, ASSET_ID, owner.clone(), amount * 2);
+		// `delegate` transfer from the `owner` without approval
+		let unapproved_result = transfer_from(
+			addr.clone(),
+			asset,
+			delegate.clone(),
+			owner.clone(),
+			BOB,
+			amount / 2,
+			vec![],
+		);
+		assert_eq!(decoded::<Error>(unapproved_result), Module { index: 52, error: 10 },);
+
+		// Check if the allowance is correct
+		assert_eq!(0, Assets::allowance(asset, &owner, &delegate.clone()));
+		assert!(
+			!approve(owner.clone(), asset, delegate.clone(), amount).did_revert(),
+			"Contract reverted!"
+		);
+		assert_eq!(Assets::allowance(asset, &owner, &delegate.clone()), amount);
```

Comment:
Only call the contract message that we are testing. Other calls should be made via the pallet directly. In addition, allowance checks are unnecessary.

Alice approves the contract to transfer on behalf of it:
```suggestion
	assert_ok!(Assets::approve_transfer(
		RuntimeOrigin::signed(ALICE.into()),
		asset_id.into(),
		addr.clone().into(),
		amount,
	));
```